library/src/main/kotlin/com/a99/rxplaces/RxAutocomplete.kt

```diff
@@ -0,0 +1,81 @@
+package com.a99.rxplaces
+
+import android.util.Log
+import android.widget.TextView
+import rx.Observable
+import rx.Scheduler
+import rx.schedulers.Schedulers
+import rx.subjects.PublishSubject
+import java.util.concurrent.TimeUnit
+
+class RxAutocomplete internal constructor(
+    val scheduler: Scheduler,
+    val repository: PlacesAutocompleteRepository,
+    val logger: (String, String) -> Unit = { _, _ -> }) {
+
+  private val autocompleteStateSubject = PublishSubject.create<AutocompleteState>()
+
+  companion object {
+    fun create(
+        apiKey: String,
+        scheduler: Scheduler = Schedulers.io(),
+        logger: (String, String) -> Unit = { tag, message -> Log.d(tag, message) }
+    ): RxAutocomplete {
+      val repository = PlacesAutocompleteRepositoryImpl(apiKey, GoogleMapsApi.create())
+      return RxAutocomplete(scheduler, repository, logger)
+    }
+
+    internal fun create(
+        scheduler: Scheduler,
+        repository: PlacesAutocompleteRepository,
+        logger: (String, String) -> Unit
+    ): RxAutocomplete {
+      return RxAutocomplete(scheduler, repository, logger)
+    }
+  }
+
+  var minKeyStroke: Int = 3
+  private var requestedTimeInterval: Long = 1500
+
+  fun setRequestedTimeInterval(interval: Long, timeUnit: TimeUnit) {
+    requestedTimeInterval = timeUnit.toMillis(interval)
+  }
+
+  fun stateStream(): Observable<AutocompleteState> {
+    return autocompleteStateSubject.onBackpressureLatest()
+  }
+
+  fun observe(
+      textView: TextView,
+      types: Array<String> = arrayOf("address"),
+      components: Array<String> = arrayOf("country:br")): Observable<List<Prediction>> {
+
+    val dataSource = RxTextView.textChanges(textView)
+        .map { it.toString() }
+
+    return observe(dataSource, types, components)
+  }
+
+  fun observe(
+      dataSource: Observable<String>,
+      types: Array<String> = arrayOf("address"),
+      components: Array<String> = arrayOf("country:br")): Observable<List<Prediction>> {
```

Comment:
Yeah, I will parametrize all configurations, but not in that PR. Is it ok for you?